vpn/vpn_dns_handler.cc

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/vpn/vpn_dns_handler.h"
+
+#include <string>
+
+#include "base/logging.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/threading/sequenced_task_runner_handle.h"
+#include "base/time/time.h"
+#include "brave/components/brave_vpn/brave_vpn_constants.h"
+#include "brave/components/brave_vpn/utils_win.h"
+#include "brave/vpn/vpn_utils.h"
+
+namespace {
+// Repeating interval to check the connection is live.
+constexpr int kCheckConnectionIntervalInSeconds = 3;
+}  // namespace
+
+namespace brave_vpn {
+
+VpnDnsHandler::VpnDnsHandler() = default;
+VpnDnsHandler::~VpnDnsHandler() {
+  CloseWatchers();
+}
+
+HANDLE VpnDnsHandler::OpenEngineSession() {
```

Comment:
what is the point of this method if it just calls `OpenWpmSession`?

Reply:
removed